instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/instrumentation.rb

```diff
@@ -85,6 +85,14 @@ def tracer
               Net::HTTP::Instrumentation.instance.tracer
             end
 
+            def filter_target(path)
+              if (filter = Net::HTTP::Instrumentation.instance.config[:filter_target])
+                filter.call(path)
+              else
+                path
+              end
+            end
```

Comment:
It is probably worthwhile protecting against the callback `raise`-ing. Should this `rescue` be more specific?
```suggestion
            rescue
              path
            end
```

Reply:
I don't know that the library should protect users from errors thrown by their filters. It's a real long-shot, but it's conceivable that someone might even _expect_ an exception to be thrown...

Reply:
debug log perhaps?

Reply:
The [error handling guidance](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#guidance) is pretty explicit:

> 1. API methods that accept external callbacks MUST handle all errors.
> 5. Beware external callbacks and overridable interfaces: Expect them to throw.

So: we should rescue and call the global error handler.

Reply:
> The [error handling guidance](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#guidance) is pretty explicit:
> 
> > 1. API methods that accept external callbacks MUST handle all errors.
> > 2. Beware external callbacks and overridable interfaces: Expect them to throw.
> 
> So: we should rescue and call the global error handler.

I stand corrected, the glorious spec is unambiguous...